blockchain/options.go

```diff
@@ -3,70 +3,94 @@ package blockchain
 import "time"
 
 const (
-	defaultEthEndpoint        = "https://rinkeby.infura.io/00iTrs5PIy0uGODwcsrb"
+	defaultLivechainEndpoint  = "https://rinkeby.infura.io/00iTrs5PIy0uGODwcsrb"
 	defaultSidechainEndpoint  = "https://sidechain-dev.sonm.com"
-	defaultGasPrice           = 20000000000 // 20 Gwei
-	defaultGasPriceSidechain  = 0
+	defaultLivechainGasPrice  = 20000000000 // 20 Gwei
+	defaultSidechainGasPrice  = 0
 	defaultBlockConfirmations = 5
+	defaultLogParsePeriod     = time.Second
 )
 
+// chainOpts describes common options
+// for almost any geth-node connection
+// (live Eth network, rinkeby, SONM sidechain
+// or local geth-node for testing).
+type chainOpts struct {
+	gasPrice           int64
+	endpoint           string
+	logParsePeriod     time.Duration
+	blockConfirmations int64
+}
+
+func (c *chainOpts) newClient() (CustomEthereumClient, error) {
+	return NewClient(c.endpoint)
+}
+
 type options struct {
-	gasPrice             int64
-	gasPriceSidechain    int64
-	apiEndpoint          string
-	apiSidechainEndpoint string
-	logParsePeriod       time.Duration
-	blockConfirmations   int64
+	livechain *chainOpts
+	sidechain *chainOpts
 }
 
 func defaultOptions() *options {
 	return &options{
-		gasPrice:             defaultGasPrice,
-		gasPriceSidechain:    defaultGasPriceSidechain,
-		apiEndpoint:          defaultEthEndpoint,
-		apiSidechainEndpoint: defaultSidechainEndpoint,
-		logParsePeriod:       time.Second,
-		blockConfirmations:   defaultBlockConfirmations,
+		livechain: &chainOpts{
```

Comment:
why options for masterchain doesn't contain `blockConfiramtions`?

Reply:
how many confirmations is applicable for masterChain settings?